packages/swapper/src/swappers/osmosis/OsmosisSwapper.ts

```diff
@@ -0,0 +1,319 @@
+    return { approvalNeeded: false }
+  }
+
+  async approveInfinite(): Promise<string> {
+    throw new Error('OsmosisSwapper: approveInfinite unimplemented')
+  }
+
+  filterBuyAssetsBySellAssetId(args: BuyAssetBySellIdInput): string[] {
+    const { sellAssetId } = args
+    if (!this.supportAssets.includes(sellAssetId)) return []
+    return this.supportAssets
+  }
+
+  filterAssetIdsBySellable(): AssetId[] {
+    return this.supportAssets
+  }
+
+  async buildTrade(args: BuildTradeInput): Promise<Trade<ChainId>> {
+    const { sellAsset, buyAsset, sellAmount } = args
+
+    if (!sellAmount) {
+      throw new Error('sellAmount is required')
+    }
+
+    const { tradeFee, rate, buyAmount } = await getRateInfo(
+      sellAsset.symbol,
+      buyAsset.symbol,
+      sellAmount !== '0' ? sellAmount : '1',
+      this.deps.osmoUrl
+    )
+
+    //convert amount to base
+    let amountBaseSell: number = parseFloat(sellAmount)
+    amountBaseSell = parseInt(String(amountBaseSell))
+    const amountBaseSellString = amountBaseSell.toString()
+
+    return {
+      buyAmount,
+      buyAsset,
+      feeData: { fee: FEE, tradeFee },
+      rate,
+      receiveAddress: '',
+      sellAmount: amountBaseSellString,
+      sellAsset,
+      sellAssetAccountNumber: 0,
+      sources: [{ name: 'Osmosis', proportion: '100' }]
+    }
+  }
+
+  async getTradeQuote(input: GetTradeQuoteInput): Promise<TradeQuote<ChainId>> {
+    const { sellAsset, buyAsset, sellAmount } = input
+    if (!sellAmount) {
+      throw new Error('sellAmount is required')
+    }
+    const { tradeFee, rate, buyAmount } = await getRateInfo(
+      sellAsset.symbol,
+      buyAsset.symbol,
+      sellAmount !== '0' ? sellAmount : '1',
+      this.deps.osmoUrl
+    )
+
+    const { minimum, maximum } = await this.getMinMax(input)
+
+    return {
+      buyAsset,
+      feeData: { fee: FEE, tradeFee },
+      maximum,
+      minimum,
+      sellAssetAccountNumber: 0,
+      rate,
+      sellAsset,
+      sellAmount,
+      buyAmount,
+      sources: DEFAULT_SOURCE,
+      allowanceContract: ''
+    }
+  }
+
+  async executeTrade({ trade, wallet }: ExecuteTradeInput<ChainId>): Promise<TradeResult> {
+    const { sellAsset, buyAsset, sellAmount, sellAssetAccountNumber, receiveAddress } = trade
+
+    const pair = sellAsset.symbol + '_' + buyAsset.symbol
+    const isFromOsmo = pair === 'OSMO_ATOM'
+    const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof SymbolDenomMapping]
+    const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof SymbolDenomMapping]
+    let ibcSellAmount
+
+    const osmosisAdapter = this.deps.adapterManager.get(osmosisChainId) as
+      | osmosis.ChainAdapter
+      | undefined
+    const cosmosAdapter = this.deps.adapterManager.get(cosmosChainId) as
+      | cosmos.ChainAdapter
+      | undefined
+
+    if (cosmosAdapter && osmosisAdapter) {
+      let sellAddress
+
+      if (!isFromOsmo) {
```

Comment:
switch on `sellAsset.assetId` to try and make this logic a bit more readable

Reply:
actually I was looking more at this and there's two types of tx's happening, from osmo and to osmo.  This change will convolute things and make it less future proof even if adding more readability when there is just two assets